### ISSUE TITLE
Bugsnag blocked by AdBlock in Firefox

### ISSUE THREAD

**Issue by andig** [bugsnag/bugsnag-js#96]
Just a heads-up: AdBlock with the EasyPrivacy filter set blocks bugsnag. Drove me nuts for hours trying to find out why bs wasn't loaded...

**Comment by ConradIrwin** [comment]
This is unfortunate, but I guess it's kind of flattering that we're now widely used enough to get noticed :/.

More concretely, we've been added to the "EasyPrivacy" blacklist https://easylist-downloads.adblockplus.org/easyprivacy.txt — which blocks loading any script called `bugsnag-*`, or any 3rd-party request to `bugsnag.com`.

There are two partial workarounds that I see:
1. Host `bugsnag-2.js` on your own server under a different name. This will allow the Bugsnag script to load so that any references to `Bugsnag` will work rather than raising an error.
2. Check for `if (Bugsnag)` before trying to use the Bugsnag object.
These won't fix notifications (as they are blocked by the other rule), but will allow your javascript that interacts with Bugsnag to keep working.

If we added js support to [bugsnag-agent](https://github.com/bugsnag/bugsnag-agent) then we could let you send notifications via your own domain without installing [Bugsnag Enterprise](https://bugsnag.com/enterprise); but to be honest, if users want to avoid being tracked enough to install something that causes as many problems as adblock, then we should probably respect their decision.

**Comment by jacobmarshall** [comment]
:+1: JavaScript support for [Bugsnag Agent](https://github.com/bugsnag/bugsnag-agent) - I'm so glad you offered to implement it @ConradIrwin :tongue:

I think implementing procedures for people to take, if privacy isn't a priority over catching bugs, is a good idea. There is no point in trying work arounds if the 'EasyPrivacy' plugin also blocks all requests to `bugsnag.com`, because whatever happens, `bugsnag-2.js` still isn't going to be able to report to `bugsnag.com`.

**Comment by ConradIrwin** [comment]
Not much we can do about this.

**Comment by btmdave** [comment]
This seems to be a bit of a problem when using in production, as many people do actually use these blockers we're finding and creates confusion.

Would be to be possible to use a cname record and post to that domain instead of bugsnag?

**Comment by jacobmarshall** [comment]
@btmdave -You could always host the Bugsnag library yourself, and make sure it's renamed to something other than `bugsnag-2.min.js`.

**Comment by btmdave** [comment]
Nice didn't realize, that works! thanks
